applications/serial_lte_modem/src/slm_cmux.c

```diff
@@ -234,26 +257,40 @@ static void cmux_starter(struct k_work *)
 int handle_at_cmux(enum at_cmd_type cmd_type)
 {
 	static struct k_work_delayable cmux_start_work;
+	const unsigned int param_count = at_params_valid_count_get(&slm_at_param_list);
+	unsigned int at_dlci;
 	int ret;
-	unsigned int op;
-	enum {
-		OP_START,
-	};
 
-	if (cmd_type != AT_CMD_TYPE_SET_COMMAND
-	 || at_params_valid_count_get(&slm_at_param_list) != 2) {
+	if (cmd_type != AT_CMD_TYPE_SET_COMMAND || param_count > 2) {
 		return -EINVAL;
 	}
 
-	ret = at_params_unsigned_int_get(&slm_at_param_list, 1, &op);
-	if (ret) {
-		return ret;
-	} else if (op != OP_START) {
-		return -EINVAL;
-	} else if (cmux_is_initialized()) {
+	if (param_count == 1 && cmux_is_started()) {
 		return -EALREADY;
 	}
 
+	if (param_count == 2) {
+		ret = at_params_unsigned_int_get(&slm_at_param_list, 1, &at_dlci);
```

Comment:
The point is for the user to be able to select which DLCI is the AT channel. So currently, when PPP is in use, which of the two channels (DLCIs) is for AT stuff and which is for PPP.

> Seems to take some extra steps for it.

What is?

Reply:
Seems like the channels are 1 and 2 for the AT-command and 0 and 1 internally. Is this really necessary?

Reply:
They are not. See [this line](https://github.com/nrfconnect/sdk-nrf/pull/14029/files#diff-9509c539610a91823494759127a5a603a355db1e6fb95428d207ff98ba81df6aR42).
In terms of CMUX the protocol DLCI 0 is always gonna be the control channel, managed and accessible only internally to the (Zephyr) CMUX module. So the AT/PPP channels are on DLCI 1 and 2.
They turn into 0 and 1 when we grab their `struct cmux_dlci` from `cmux.dlcis`, which is an array, and so DLCI 1 being the first channel (to us), an offset conversion of one has to be made.
If this is too confusing there could be a function added that returns a `struct cmux_dlci` by passing it the DLCI (address). Though conversion the other way around would be needed as well I think, so that would be two functions.